### DISCUSSION TITLE
Adding power (toggle) switch for HAT board power input.

### DISCUSSION THREAD

**Discussion by Jkh62** [jgyates/genmon#1034] [Questions]
I am considering wiring in a manual toggle switch to cut power / reset the pi inside its enclosure. Pin #1 on the Generac Molex supplies the 5V to the pi & HAT board ( I think from memory).  Nipping the lug behind the Molex connector and soldering leads to a toggle switch I can mount inside the case is simple to do . I am thinking out loud and really do not know if its needed or if the pi runs solid and never needs a manual power cycle to reset / reboot it. Its easier to open the door and flip a switch on and off Vs unplugging a cable.

Am I good to go powering my pi zero W from the Molex cable or is it better to use the stand alone power cable from the battery ?
Generac 22KW with evo2 controller.

**Comment by skipfire** [comment]
The PintSize HATs are powered by 12v regardless of them being run over the power cable or the Molex cable.  You could insert a switch into either cable, or if you use the power cable you could buy a small extension switch like this one from Amazon https://www.amazon.com/dp/B0CLT2Z8MJ.  The leads are a bit short though, so you could add another extension after it to allow the length you want for routing it to wherever you have your Pi.  That said, the first time I had to go and touch my Pi after installing it was after a lightning strike made it unresponsive, but it also fried some stuff on the Pi so I don't think that really counts.  I won't say it would never need a manual touch, but it should be infrequent.  I believe the more likely event is needing to replace the micro-SD card in a couple years, I've found that the Pi becoming unresponsive and needing a manual reset is one of the first indications of a micro-SD card failing.

As for powering via Molex or cable, you can try with the Molex, but if it draws too much power the controller will think the utility voltage is low and try to start the generator, or if in manual mode just display the low utility condition.  If that happens you will have to use the power cable.  I don't think I've heard of any Pi Zero Ws triggering that condition, however some Pi Zero 2s have, a larger number of 3A+ and 3Bs, and all 3B+, 4s, and 5s seem to trigger that condition.

**Comment by Jkh62** [comment]
Does not sound like I need an added power switch but the SD writes are a real item to keep tabs on.  Lightning does not make anything better and really plays hell on electronics.  Thx for the Amazon link.
